compiler/rustc_mir_transform/src/check_packed_ref.rs

```diff
@@ -37,10 +37,10 @@ fn unsafe_derive_on_repr_packed(tcx: TyCtxt<'_>, def_id: LocalDefId) {
         // FIXME: when we make this a hard error, this should have its
         // own error code.
         let message = if tcx.generics_of(def_id).own_requires_monomorphization() {
-            "`#[derive]` can't be used on a `#[repr(packed)]` struct with \
+            "this trait can't be derived on this `#[repr(packed)]` struct with \
```

Comment:
> It would be better to replace "this trait can't be derived..." with "Debug can't be derived..." (or whatever trait name is appropriate), but I couldn't work out how to get the trait name. Suggestions are welcome!

I'm pretty sure the `def_id` here corresponds to `Debug`, so you can use `tcx.item_name(def_id)` to modify the message.

Reply:
This wasn't enough, returning `None`. But this worked:
```rust
tcx.item_name(tcx.trait_id_of_impl(def_id.to_def_id()).expect("derived trait name"))
```